### PR TITLE
libplugin: don't call callbacks if cmd completed before response, and fix another commando crash

### PR DESCRIPTION
**Pull request by rustyrussell**
The callbacks-after-command-completed can particularly happen with commando:

```
 commando: FATAL SIGNAL 11 (version 06b36d3)
0x55609e953d51 send_backtrace
	common/daemon.c:33
0x55609e953dfb crashdump
	common/daemon.c:46
0x7f665e3b908f ???
	/build/glibc-SzIz7B/glibc-2.31/signal/../sysdeps/unix/sysv/linux/x86_64/sigaction.c:0
0x55609e9387a3 send_more_cmd
	plugins/commando.c:632
0x55609e93b270 handle_rpc_reply
	plugins/libplugin.c:669
0x55609e93bd50 rpc_read_response_one
	plugins/libplugin.c:842
0x55609e93be86 rpc_conn_read_response
	plugins/libplugin.c:862
0x55609e9f4f68 next_plan
	ccan/ccan/io/io.c:59
0x55609e9f5b70 do_plan
	ccan/ccan/io/io.c:407
0x55609e9f5bb2 io_ready
	ccan/ccan/io/io.c:417
0x55609e9f7ea5 io_loop
	ccan/ccan/io/poll.c:453
0x55609e93eb20 plugin_main
	plugins/libplugin.c:1676
0x55609e9397ab main
	plugins/commando.c:922
0x7f665e39a082 __libc_start_main
	../csu/libc-start.c:308
0x55609e93677d ???
	???:0
0xffffffffffffffff ???
	???:0
```

Reported-by: @adi2011
Changelog-None

### CODE REVIEW

**Comment by adi2011** [comment]
ACK d778b2c

**Review by vincenzopalazzo — APPROVED** [review]
ACK https://github.com/ElementsProject/lightning/pull/5442/commits/d778b2c3aea43e8ee44fe2f342a99003f8060713

**Comment by rustyrussell** [comment]
We shouldn't ignore that crash! There's something else going on ..

**Review by vincenzopalazzo — APPROVED** [review]
ACK https://github.com/ElementsProject/lightning/pull/5442/commits/c04c195ff90d8e4dffd5bb93c4c9316e6a4c93d1

The CI catch two crash, the "unrelated failure" is only a distant memory

File: contrib/pyln-testing/pyln/testing/utils.py [hunk 1]
```diff
@@ -200,24 +200,29 @@ def __init__(self, outputDir, verbose=True):
...
-            self.proc = subprocess.Popen(self.cmd_line,
-                                         stdin=stdin,
-                                         stderr=self.stderr_write,
-                                         env=self.env)
+            stdout = None
+        if stderr_redir:
+            stderr = self.stderr_write
+            self.stderr_redir = True
+        else:
+            stderr = None
```
Comment (cdecker) [inline contrib/pyln-testing/pyln/testing/utils.py:218]: Any reason we still open the `self.stderr_write` and `self.stderr_read` if we never intend to use them?

**Review by cdecker — APPROVED** [review]
Looking good so far :-)

ACK c04c195ff90d8e4dffd5bb93c4c9316e6a4c93d1

**Comment by niftynei** [comment]
ACK c04c195ff90d8e4dffd5bb93c4c9316e6a4c93d1

**Pull request merged**